torch/csrc/autograd/function.cpp

```diff
@@ -17,7 +17,13 @@
 
 namespace torch { namespace autograd {
 
-thread_local uint64_t Function::next_sequence_nr_ = 0;
+/// Monotonically incrementing (thread local!) counter to supply sequence
+/// numbers.
+thread_local uint64_t Function_next_sequence_nr_ = 0;
```

Comment:
Why can't the thread local variables be class-static? Why do they have to be global? Windows?

Reply:
Yes, windows. See my point 2 in my big comment above in the main thread.

Specifically the restriction here isn't about class-static vs global, but rather that it can't be in the header (and this is a smaller change than moving the entire class out of the header)

Reply:
Also, to be clear, it is not *static* variables that cause the problem, it is *thread-local* variables. (And if you API a class, it implicitly APIs all of the static variables in it, which is the proximal problem).

This change is fine, as far as I'm concerned.

Reply:
++